### DISCUSSION TITLE
Epoptes problems after upgrading to ubuntu mate 20.04

### DISCUSSION THREAD

**Discussion by nektarios1201** [epoptes/epoptes#135] [Q&A]
After upgrading both client and server to ubuntu mate 20.04 from 18.04 I have the following problems.
1. I cannot turn on the 20.04 machine remotely, although  I can restart and turn it off. At the same time I can turn on  a machine with 18.04 that have not upgrade yet. 
2. When i run the command  _sudo epoptes-client -c_ from either client machine (20.04 or 18.04) I get the following error
_139623726707136:error:1408F10B:SSL routines:ssl3_get_record:wrong version number:../ssl/record/ssl3_record.c:332:
epoptes-client ERROR: Failed to fetch certificate from server:789_

I have to mention that before upgrading everything was ok, plus that apart from the above problems, epoptes work fine.
Thanks 
Nektarios kokolakis

**Comment by nektarios1201** [comment]
I solved the first problem, it had nothing to do with epotes. It seems upgrading disable wol on some machines not all and also removed mac address from the device name of the connection. So run _sudo ethtool -s <NIC> wol g_ to enable wol and add mac address to device name of the interface.
Second problem still remain.

Thanks and sorry I didn't find  this before write the first problem

**Comment by alkisg** [comment]
The epoptes-client script has this line:
ethtool -s "$def_iface" wol $WOL

So if it's properly installed, it automatically enables WOL, you don't need to do it manually.
So I guess the first problem was caused by the second problem.

Now regarding the second problem, what's the output of this command on the server?
ls -l /etc/epoptes

**Reply by nektarios1201** [reply]
σύνολο 4
-rw-r--r-- 1 root root   0 Ιουν  3  2020 server.crt
-rw-r--r-- 1 root root 920 Ιουν  3  2020 server.key

**Comment by alkisg** [comment]
Your certificate is empty (zero size), that's why epoptes isn't working.
This means that you ran `epoptes-client -c` while on the server itself, so you tried to fetch the certificate and overwrite it at the same time.

The easiest way to recreate the certificate is to run:

```shell
sudo rm /etc/epoptes/*
sudo dpkg-reconfigure epoptes
```

Then send the new certificate to your clients as usual.

**Reply by nektarios1201** [reply]
Thanks very much.
Problem solved. On Monday,  I will upgrade another computer to find out  if solving problem 2, will solve problem 1 as well.
